mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -175,6 +178,9 @@ public void setInMailboxes(MessageId messageId, List<MailboxId> mailboxIds, Mail
             List<MailboxId> mailboxesToRemove = ImmutableList.copyOf(Sets.difference(currentMailboxes, targetMailboxes));
             SetView<MailboxId> mailboxesToAdd = Sets.difference(targetMailboxes, currentMailboxes);
 
+            assertRightsOnMailboxes(mailboxesToAdd, mailboxSession, MailboxACL.Right.Insert);
+            assertRightsOnMailboxes(mailboxesToRemove, mailboxSession, MailboxACL.Right.DeleteMessages);
```

Comment:
I really think this method was already too long before these modifications. Please refactor it to make it shorter

Reply:
I gave it a try but it's not as good as I'd like.